driver/docker-container/factory.go

```diff
@@ -39,14 +41,20 @@ func (f *factory) New(ctx context.Context, cfg driver.InitConfig) (driver.Driver
 	}
 	d := &Driver{factory: f, InitConfig: cfg}
 	for k, v := range cfg.DriverOpts {
-		switch k {
-		case "network":
+		switch {
+		case k == "network":
 			d.netMode = v
 			if v == "host" {
 				d.InitConfig.BuildkitFlags = append(d.InitConfig.BuildkitFlags, "--allow-insecure-entitlement=network.host")
 			}
-		case "image":
+		case k == "image":
 			d.image = v
+		case strings.HasPrefix(k, "env."):
+			envName := k[4:]
+			if len(envName) == 0 {
```

Comment:
nit: `envName == ""`

Reply:
Fixed in 046a3b0939a7c7b2c036697a44380b4d7f55151f.